src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -775,12 +772,32 @@ private static SortedSet<string> GetCountersForDocument(BlittableJsonReaderArray
             return countersToAdd;
         }
 
+        private static void ValidateCounterName(string documentId, string counterName)
+        {
+            if (counterName.Length > DocumentIdWorker.MaxIdSize)
+            {
+                ThrowCounterNameTooBig(counterName);
+            }
+
+            if (documentId.Length + counterName.Length > DocumentIdWorker.MaxIdSize)
+            {
+                ThrowStorageKeyTooBig(documentId, counterName);
+            }
+        }
+
         private static void ThrowCounterNameTooBig(string name)
         {
             throw new ArgumentException(
-                $"Counter name cannot exceed {MaxCounterNameSize} bytes, but counter name was {Encoding.UTF8.GetByteCount(name)} bytes. The invalid counter name is '{name}'.",
+                $"Counter name cannot exceed {DocumentIdWorker.MaxIdSize} bytes, but counter name was {Encoding.UTF8.GetByteCount(name)} bytes. The invalid counter name is '{name}'.",
```

Comment:
I think we can get rid of the GetByteCount here? We are using string.Length for that now?

Reply:
done